### ISSUE TITLE
Slow multiply

### ISSUE THREAD

**Issue by shawnl** [SpinalHDL/VexRiscv#132]
If the Cortex-M0 can have a slow multiply option (32 cycles I believe), why can't the VexRiscv have it to save space? (I am not sure if there is a multiply only subset of the M option, like the bitmanip stuff has many configurations) Thanks.

**Comment by tomverbeure** [comment]
The VexRiscv can have it, is just that nobody has needed it so far. 

Most FPGAs have HW multipliers/DSPs, often with built-in pipeline FFs. So on an FPGA, a DSP-based multiplier will cost the same (or less!) than a slow multiplier.

The RISCV specification doesn't have an option that has a MUL but not a DIV. However, there is an option in GCC that forces it to not generate DIVs, even with the M option is specified.

**Comment by Dolu1990** [comment]
Agree, as most VexRIscv targets are FPGA with DSP blocks, it isn't very much used. but it is already implemented.
Basicaly, instead of using the new MulSimplePlugin + new DivPlugin in the configuration, you have to use : 

```scala
          new MulDivIterativePlugin(
            genMul = true,
            genDiv = true,
            mulUnrollFactor = 1,
            divUnrollFactor = 1
          ),
``` 

It will implement mul/div using no DSP blocks, but bit by bit (about 34 cycles for one mul).
Also, you can get faster mul by setting mulUnrollFactor = 4, which will do 4 bits per clock instead of one.

**Comment by tomverbeure** [comment]
Oops! I should have known better! :-)

**Comment by Dolu1990** [comment]
No worries XD
